### PR TITLE
Add custom markdownlint config and update docs to be markdownlint clean

### PR DESCRIPTION
**Pull request by scramsby**
* Add custom markdownlint config.
* Fixes docs to be markdownlint clean (see https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md).
  * Fix headings to start at #.
  * Remove redundant whitespace.

### CODE REVIEW

File: docs/IWYUMappings.md [hunk 1]
```diff
@@ -102,12 +97,11 @@ For example;
 
 The rationale for the `ref` directive was to make it easier to compose project-specific mappings from a set of library-oriented mapping files. For example, IWYU might ship with mapping files for [Boost](http://www.boost.org), the SCL, various C standard libraries, the Windows API, the [Poco Library](http://pocoproject.org), etc. Depending on what your specific project uses, you could easily create an aggregate mapping file with refs to the relevant mappings.
 
-
-#### Specifying Mapping Files ####
+### Specifying Mapping Files ###
 
 Mapping files are specified on the command-line using the `--mapping_file` switch:
 
-    $ include-what-you-use -Xiwyu --mapping_file=foo.imp some_file.cc
```
Comment (kimgr) [inline docs/IWYUMappings.md]: I like the `$` as a command-line prompt indication, can we please keep that?

File: docs/IWYUPragmas.md [hunk 2]
```diff
@@ -1,23 +1,21 @@
-## IWYU pragmas ##
+# IWYU pragmas #
 
 IWYU pragmas are used to give IWYU information that isn't obvious from the source code, such as how different files relate to each other and which includes to never remove or include.
 
-All pragmas start with `// IWYU pragma: ` or `/* IWYU pragma: `. They are case-sensitive and spaces are significant.
```
Comment (kimgr) [inline docs/IWYUPragmas.md]: The trailing spaces in the pragma examples were intended, that's really the pattern that pragmas start with.

File: docs/WhatIsAUse.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-## What Is a Use? ##
+# What Is a Use #
```
Comment (kimgr) [inline docs/WhatIsAUse.md]: I'd like to keep this question mark

File: docs/WhyIWYU.md [hunk 4]
```diff
@@ -1,48 +1,42 @@
-## Why Include What You Use? ##
```
Comment (kimgr) [inline docs/WhyIWYU.md]: Keep the ?

File: docs/WhyIWYU.md [hunk 5]
```diff
@@ -1,48 +1,42 @@
...
-
-### Dependency Cutting ###
+## Dependency Cutting ##
 
 Again, this makes the most sense for large code-bases.  Suppose your binaries are larger than you would expect, and upon closer examination use symbols that seem totally irrelevant.  Where do they come from?  Why are they there?  With include-what-you-use, you can easily determine this by seeing who includes the files that define these symbols: those includers, and those alone, are responsible for the use.
 
 Once you know where a symbol is used in your binary, you can see how practical it is to remove that use, perhaps by breaking up the relevant .h files into two parts, and fixing up all callers.  Again it's IWYU to the rescue: with include-what-you-use, figuring out the callers that need fixing is easy.
 
-
-### Why Forward-Declare? ###
```
Comment (kimgr) [inline docs/WhyIWYU.md]: Keep the ?

File: docs/WhyIWYUIsDifficult.md [hunk 6]
```diff
@@ -136,28 +129,25 @@ Another case is a header file like this:
...
 
 Conditional `#includes` are a problem for IWYU when the condition is false:
 
     #if _MSC_VER
     #include <foo>
     #endif
 
-If we're not running under windows (and IWYU does not currently run under windows), we have no way of telling if foo is a necessary `#include` or not.
-
+Since we're not running using the Microsoft Visual C++ compiler (IWYU is built on top of Clang), we have no way of telling if foo is a necessary `#include` or not.
```
Comment (kimgr) [inline docs/WhyIWYUIsDifficult.md]: I'd like for this to be a separate commit from all the formatting changes, as it makes content changes

File: docs/IWYUPragmas.md (hunk 2 above)
Comment (scramsby) [reply docs/IWYUPragmas.md]: Okay. Those trailing spaces are omitted in the generated HTML when using backticks for inline HTML. I've thus changed to use <code> tags in these instances.

File: docs/WhatIsAUse.md (hunk 3 above)
Comment (scramsby) [reply docs/WhatIsAUse.md]: Done. Added an exception for this rule.

File: docs/WhyIWYU.md (hunk 4 above)
Comment (scramsby) [reply docs/WhyIWYU.md]: Done. Added an exception for this rule.

File: docs/WhyIWYU.md (hunk 5 above)
Comment (scramsby) [reply docs/WhyIWYU.md]: Done. Added an exception for this rule.

File: docs/IWYUMappings.md (hunk 1 above)
Comment (scramsby) [reply docs/IWYUMappings.md]: Done. Added an exception for this rule.

File: docs/WhyIWYUIsDifficult.md (hunk 6 above)
Comment (scramsby) [reply docs/WhyIWYUIsDifficult.md]: Sure.

File: docs/IWYUPragmas.md (hunk 2 above)
Comment (kimgr) [reply docs/IWYUPragmas.md]: Hm, that screws up the Markdown for the sake of one renderer. I'd like for the Markdown source to be copy/paste-friendly and readable. Let's put this back the way it was.

File: .markdownlint.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+    "default": true,
+    "MD013": false,
+    "MD014": false,
+    "MD026": false,
+    "MD033": { "allowed_elements": ["code"] }
+}
```
[the remark this answers is not in the document]
Comment (scramsby) [reply .markdownlint.json]: Sure.

**Comment by scramsby** [comment]
Which renderer actually does render the trailing space as intended when between backticks? It's not copy/paste friendly from the rendered result on GitHub (which is I imagine the most common viewing place) and it's almost the same copy/paste friendly from the markdown source (between backticks versus between <code> markers.

**Comment by kimgr** [comment]
I meant from the Markdown source. The `&nbsp;` is not appreciated by more primitive tools.

When I rendered a PDF using pandoc, `<code>` tags weren't recognized and the text was just body-text-styled, and I'd prefer if HTML wasn't the only conceivable target.

**Comment by scramsby** [comment]
Gotcha. Will just revert then.

File: docs/IWYUPragmas.md [hunk 8]
```diff
@@ -169,22 +161,21 @@ You can explicitly mark an arbitrary `#include` directive as denoting the associ
 
 You can mark multiple `#include` directives as associated and they will all be considered as such.
 
-
-### Which pragma should I use? ###
+## Which pragma should I use ##
```
Comment (kimgr) [inline docs/IWYUPragmas.md]: Oops, another question mark disappeared

File: docs/WhyIWYUIsDifficult.md [hunk 9]
```diff
@@ -54,8 +52,7 @@ Even normal template arguments can be confusing. Consider this templated functio
 
 and you call `MyFunc(FunctionReturningAFunctionPointer())`. What types are being used where, in this case?
 
-
-### Who is Responsible for Dependent Template Types? ###
+## Who is Responsible for Dependent Template Types ##
```
Comment (kimgr) [inline docs/WhyIWYUIsDifficult.md]: Apparently we have a thing for question marks in titles :)

**Comment by kimgr** [comment]
The build failure appears to be some temporary apt lapse. Let's merge this, it looks good to me. Thanks for the re-take after my merge snafu.
